src/bucket.js

```diff
@@ -2297,20 +2297,21 @@ class Bucket extends common.ServiceObject {
     if (is.boolean(options.resumable)) {
       upload();
     } else if (isURL) {
-      request.head(requestOptions, (err, resp) => {
-        if (err) {
-          callback(err);
-          return;
-        }
-
-        const contentLength = resp.headers['content-length'];
-
-        if (is.number(contentLength)) {
-          options.resumable = contentLength > RESUMABLE_THRESHOLD;
-        }
+      const url = requestOptions.url;
+      const method = 'HEAD';
+      const headers = requestOptions.headers;
+      const redirect = requestOptions.followAllRedirects ? 'follow' : 'manual';
+      fetch(url, {method, headers, redirect})
```

Comment:
I do not believe fetch will follow proxies by default :(  `request` does this for us, and it's one of the reasons `axios` just isn't going to work out.  I am starting to think we should create a `google-http-client` npm module that wraps our retry logic, backoff, proxy support, and redirect settings in a single place.

Reply:
Yes to putting it in a library, but this one doesn't do retry or backoff. Let's keep it simple. It's fetching images form somewhere not GCP related, no need to add a bunch of logic.

Reply:
For retry/backoff in this case, that may be fine.  The problem is proxies :(